src/main/java/com/amazon/ion/impl/macro/MacroEvaluator.kt

```diff
@@ -52,6 +54,8 @@ class MacroEvaluator {
         private var numExpandedExpressions = 0
         /** Pool of [ExpansionInfo] to minimize allocation and garbage collection. */
         private val expanderPool: ArrayList<ExpansionInfo> = ArrayList(32)
+        /** Pool of [Expression] to minimize allocation and garbage collection. */
+        val expressionPool: PooledExpressionFactory = PooledExpressionFactory()
```

Comment:
Note: this means there are two expression pools: one used when parsing expression arguments, and one used when evaluating expressions (which may produce more expressions). I think this is fine, as both pools are one-time costs, and the shape of the expressions they hold may be very different.